sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/samples/Readme.md

```diff
@@ -23,3 +37,482 @@ var client = new TimeSeriesInsightsClient(
+    }
+    else
+    {
+        Console.WriteLine($"Replaced Time Series Insights instance with Id '{tsiId}'.");
+    }
+}
+```
+
+### Delete instances
+
+To delete instances, pass in a list of time series Ids or names for the instances you want to delete.
+```C# Snippet:TimeSeriesInsightsSampleDeleteInstanceById
+var instancesToDelete = new List<TimeSeriesId>
+{
+    instanceId,
+};
+
+Response<TimeSeriesOperationError[]> deleteInstanceErrors = await client
+    .Instances
+    .DeleteAsync(instancesToDelete)
+    .ConfigureAwait(false);
+
+// The response of calling the API contains a list of error objects corresponding by position to the input parameter
+// array in the request. If the error object is set to null, this means the operation was a success.
+for (int i = 0; i < deleteInstanceErrors.Value.Length; i++)
+{
+    TimeSeriesId tsiId = instancesToDelete[i];
+
+    if (deleteInstanceErrors.Value[i] == null)
+    {
+        Console.WriteLine($"Deleted Time Series Insights instance with Id '{tsiId}'.");
+    }
+    else
+    {
+        Console.WriteLine($"Failed to delete a Time Series Insights instance with Id '{tsiId}'. Error Message: '{deleteInstanceErrors.Value[i].Message}'");
+    }
+}
+```
+
+## Create, get, replace, and delete types
+
+### Create types
+You can use the following snippet to create types.
+
+```C# Snippet:TimeSeriesInsightsSampleCreateType
+// Create an aggregate type
+var timeSeriesTypes = new List<TimeSeriesType>();
+
+var countExpression = new TimeSeriesExpression("count()");
+var aggregateVariable = new AggregateVariable(countExpression);
+var variables = new Dictionary<string, TimeSeriesVariable>();
+var variableName = "aggregateVariable";
+variables.Add(variableName, aggregateVariable);
+
+var timeSeriesTypesProperties = new Dictionary<string, string>
+{
+    { "Type1", "Type1Id"},
+    { "Type2", "Type2Id"}
+};
+
+foreach (KeyValuePair<string, string> property in timeSeriesTypesProperties)
+{
+    var type = new TimeSeriesType(property.Key, variables)
+    {
+        Id = property.Value
+    };
+    timeSeriesTypes.Add(type);
+}
+
+Response<TimeSeriesTypeOperationResult[]> createTypesResult = await client
+    .Types
+    .CreateOrReplaceAsync(timeSeriesTypes)
+    .ConfigureAwait(false);
+
+// The response of calling the API contains a list of error objects corresponding by position to the input parameter array in the request.
+// If the error object is set to null, this means the operation was a success.
+for (int i = 0; i < createTypesResult.Value.Length; i++)
+{
+    if (createTypesResult.Value[i].Error == null)
+    {
+        Console.WriteLine($"Created Time Series type successfully.");
+    }
+    else
+    {
+        Console.WriteLine($"Failed to create a Time Series Insights type: {createTypesResult.Value[i].Error.Message}.");
+    }
+}
+```
+
+### Get types
+
+Using `Types.GetTypesAsync`, all created types or errors are returned as `AsyncPageable<TimeSeriesType>`.
```

Comment:
is this true? doesn't GetTypesAsync only returns a list of TimeSeriesType, not errors?